From 70e36b1a4e67a162c44a692bcde6bec6b6bdc9ee Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@redhat.com>
Date: Tue, 14 Mar 2017 22:55:27 -0400
Subject: [PATCH 72/75] Add s390/s390x --info test (#1285601)

Running the --info grubby option on s390/s390x platforms always returned
the (null) value. Added a test to look at --info and ensure it comes out
as a user would expect.

Resolves: rhbz#1285601
---
 test.sh                |  2 ++
 test/results/info/z2.1 |  6 ++++++
 test/zipl.2            | 22 ++++++++++++++++++++++
 3 files changed, 30 insertions(+)
 create mode 100644 test/results/info/z2.1
 create mode 100644 test/zipl.2

diff --git a/test.sh b/test.sh
index 3631ed0..34ba4d7 100755
--- a/test.sh
+++ b/test.sh
@@ -320,6 +320,8 @@ testing="Z/IPL display default title"
 ziplDisplayTest zipl.1 defaulttitle/z.1 --default-title
 testing="Z/IPL display default kernel"
 ziplDisplayTest zipl.1 defaultkernel/z.1 --default-kernel
+testing="Z/IPL display entry information"
+ziplDisplayTest zipl.2 info/z2.1 --info=1
 
 testing="GRUB fallback directive"
 grubTest grub.5 fallback/g5.1 --remove-kernel=/boot/vmlinuz-2.4.7-ac3 \
diff --git a/test/results/info/z2.1 b/test/results/info/z2.1
new file mode 100644
index 0000000..e5ce3a1
--- /dev/null
+++ b/test/results/info/z2.1
@@ -0,0 +1,6 @@
+index=1
+kernel=/boot/vmlinuz-3.10.0-514.6.2.el7.s390x
+args="crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0 LANG=en_US.UTF-8 systemd.log_level=debug systemd.log_target=kmsg"
+root=/dev/mapper/rhel_ibm--z--68-root
+initrd=/boot/initramfs-3.10.0-514.6.2.el7.s390x.img
+title=3.10.0-514.6.2.el7.s390x_with_debugging
diff --git a/test/zipl.2 b/test/zipl.2
new file mode 100644
index 0000000..f064b2a
--- /dev/null
+++ b/test/zipl.2
@@ -0,0 +1,22 @@
+[defaultboot]
+defaultauto
+prompt=1
+timeout=5
+default=3.10.0-514.6.2.el7.s390x_with_debugging
+target=/boot
+[3.10.0-514.6.2.el7.s390x]
+	image=/boot/vmlinuz-3.10.0-514.6.2.el7.s390x
+	parameters="root=/dev/mapper/rhel_ibm--z--68-root crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0 LANG=en_US.UTF-8"
+	ramdisk=/boot/initramfs-3.10.0-514.6.2.el7.s390x.img
+[3.10.0-514.6.2.el7.s390x_with_debugging]
+	image=/boot/vmlinuz-3.10.0-514.6.2.el7.s390x
+	parameters="root=/dev/mapper/rhel_ibm--z--68-root crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0 LANG=en_US.UTF-8 systemd.log_level=debug systemd.log_target=kmsg"
+	ramdisk=/boot/initramfs-3.10.0-514.6.2.el7.s390x.img
+[linux]
+	image=/boot/vmlinuz-3.10.0-514.el7.s390x
+	ramdisk=/boot/initramfs-3.10.0-514.el7.s390x.img
+	parameters="root=/dev/mapper/rhel_ibm--z--68-root crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0 LANG=en_US.UTF-8"
+[linux-0-rescue-196fd0394b7442a6838686745f03fd04]
+	image=/boot/vmlinuz-0-rescue-196fd0394b7442a6838686745f03fd04
+	ramdisk=/boot/initramfs-0-rescue-196fd0394b7442a6838686745f03fd04.img
+	parameters="root=/dev/mapper/rhel_ibm--z--68-root crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0"
-- 
2.9.3

